src/main/java/com/dreammaster/gthandler/multiAirFilter/GT_MetaTileEntity_AirFilterBase.java

```diff
@@ -3,95 +3,135 @@
-			return tt.getInformation();
-		} else {
-			return tt.getStructureInformation();
-		}  	
+        final GT_Multiblock_Tooltip_Builder tt = new GT_Multiblock_Tooltip_Builder();
+        tt.addMachineType("Air Filter")
+                .addInfo("Controller block for the Electric Air Filter T"+ multiTier)
+                .addInfo("Needs a Turbine in the controller")
+                .addInfo("Add " + ItemList.AdsorptionFilter.getIS().getDisplayName() + " in input bus to double efficiency (30 uses per item)")
+                .addInfo("Machine tier = Maximum effective Muffler tier")
+                .addInfo("Can process "+(2*multiTier+1)+"x"+(2*multiTier+1)+" chunks")
+                .addInfo("Each muffler reduce pollution by 30 * TurbineEfficiency * Floor(2.5^Tier) every second")
+                .addSeparator()
+                .beginStructureBlock(3, 4, 3, true)
+                .addController("Front bottom")
+                .addOtherStructurePart(GT_LanguageManager.getTranslation(GT_Container_CasingsNH.sBlockCasingsNH.getUnlocalizedName()+"."+getCasingMeta()+".name"), "Top and bottom")
+                .addOtherStructurePart(GT_LanguageManager.getTranslation(GT_Container_CasingsNH.sBlockCasingsNH.getUnlocalizedName()+"."+getPipeMeta()+".name"), "Corners of the middle 2 layers")
+                .addOtherStructurePart("Muffler Hatch Or "+GT_LanguageManager.getTranslation(GT_Container_CasingsNH.sBlockCasingsNH.getUnlocalizedName()+"."+getPipeMeta()+".name"), "8 in the middle layers")
+                .addEnergyHatch("Any bottom layer casing")
+                .addMaintenanceHatch("Any bottom layer casing")
+                .addInputBus("Any bottom layer casing")
+                .addOutputBus("Any bottom layer casing")
+                .toolTipFinisher("GTNH Coremod");
+        if (!Keyboard.isKeyDown(Keyboard.KEY_LSHIFT)) {
+            return tt.getInformation();
+        } else {
+            return tt.getStructureInformation();
+        }
     }
 
     @Override
     public ITexture[] getTexture(IGregTechTileEntity aBaseMetaTileEntity, byte aSide, byte aFacing, byte aColorIndex, boolean aActive, boolean aRedstone) {
         if (aSide == aFacing) {
-            return new ITexture[]{Textures.BlockIcons.getCasingTextureForId(57), new GT_RenderedTexture(aActive ? Textures.BlockIcons.OVERLAY_FRONT_DIESEL_ENGINE_ACTIVE : Textures.BlockIcons.OVERLAY_FRONT_DIESEL_ENGINE)};
+            return new ITexture[]{Textures.BlockIcons.getCasingTextureForId(getCasingIndex()), new GT_RenderedTexture(aActive ? Textures.BlockIcons.OVERLAY_FRONT_DIESEL_ENGINE_ACTIVE : Textures.BlockIcons.OVERLAY_FRONT_DIESEL_ENGINE)};
+        }
+        return new ITexture[]{Textures.BlockIcons.getCasingTextureForId(getCasingIndex())};
+    }
+
+    public int getCasingIndex(){
+        if (multiTier <=1){
+            return 57;
+        }
+        else if (multiTier == 2){
+            return 59;
+        }
+        else{
+            return 60;
```

Comment:
"magic numbers", would suggest extracting as constants to class which is responsible on tier description